In [1]:
import pandas as pd

In [2]:
# use r before " when we get error for version......have the habbit for using r
cr = pd.read_csv(r"O:\IMARTICUS LECTURES\Python\Pandas Data Set\CreditRisk.csv")
cr

Loan_ID Gender Married  Dependents     Education Self_Employed  \
0    LP001002   Male      No         0.0      Graduate            No   
1    LP001003   Male     Yes         1.0      Graduate            No   
2    LP001005   Male     Yes         0.0      Graduate           Yes   
3    LP001006   Male     Yes         0.0  Not Graduate            No   
4    LP001008   Male      No         0.0      Graduate            No   
..        ...    ...     ...         ...           ...           ...   
976  LP002971   Male     Yes         4.0  Not Graduate           Yes   
977  LP002975   Male     Yes         0.0      Graduate            No   
978  LP002980   Male      No         0.0      Graduate            No   
979  LP002986   Male     Yes         0.0      Graduate            No   
980  LP002989   Male      No         0.0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         NaN             360.0   
1               4583             1508.0       128.0             360.0   
2               3000                0.0        66.0             360.0   
3               2583             2358.0       120.0             360.0   
4               6000                0.0       141.0             360.0   
..               ...                ...         ...               ...   
976             4009             1777.0       113.0             360.0   
977             4158              709.0       115.0             360.0   
978             3250             1993.0       126.0             360.0   
979             5000             2393.0       158.0             360.0   
980             9200                0.0        98.0             180.0   

     Credit_History Property_Area Loan_Status  
0               1.0         Urban           Y  
1               1.0         Rural           N  
2               1.0         Urban           Y  
3               1.0         Urban           Y  
4               1.0         Urban           Y  
..              ...           ...         ...  
976             1.0         Urban           Y  
977             1.0         Urban           Y  
978             NaN     Semiurban           Y  
979             1.0         Rural           N  
980             1.0         Rural           Y  

[981 rows x 13 columns]

In [23]:
# using cross tab function on columns

# Problem 1-->

# is education impacting loan_status yes or no
# yes then why?
# no then why?


# Ans - Yes Eduaction has a impact on Loan_status
#        - There is 74.31 % of chances of getting loan for person who is graduate
#         and there is only 66.51 % chances of getting loan for person who is not graduate

# so here the advantage is for graduate cutomer for closing the loan


In [24]:
pd.crosstab(cr.Loan_Status, cr.Education)

Education    Graduate  Not Graduate
Loan_Status                        
N                 196            73
Y                 567           145

In [25]:
# for graduates

(567 /(196 + 567) * 100)

74.31192660550458

In [26]:
# for non - graduates

(145 /(73 + 145) * 100)


66.5137614678899

In [ ]:
# Problem 2-->

# is Credit_history impacting loan_status yes or no
# yes then why?
# no then why?

# Ans --> it shows that there are 84.88% of chances 
#         for loan approval if the credit history is 1
# and 
#       - there is 94.59% of changes of rejecting the loan when credit_history is 0

In [28]:
pd.crosstab(cr.Credit_History , cr.Loan_Status)

Loan_Status       N    Y
Credit_History          
0.0             140    8
1.0             114  640

In [38]:
# for credit history 1 (Yes)

640/(114+640) * 100

84.88063660477454

In [41]:
# for rejection

140/(8+140) * 100

94.5945945945946

In [42]:
cr.ApplicantIncome.describe(percentiles = [.1,.2,.3,.4,.5,.6,.7,.8,.9,1])

count      981.000000
mean      5179.795107
std       5695.104533
min          0.000000
10%       2221.000000
20%       2609.000000
30%       3062.000000
40%       3413.000000
50%       3800.000000
60%       4301.000000
70%       5000.000000
80%       6080.000000
90%       8750.000000
100%     81000.000000
max      81000.000000
Name: ApplicantIncome, dtype: float64

Applicant income -->
what is 30% ---> of 981 people are earning less than or equal to 3062 USD per month